cantusdb/turtle/csv2rdf_by_RDFLib.py

```diff
@@ -3,64 +3,60 @@
 
 # helper to get the predicate relation dictionary
 def get_ontology_dict () :
-    
-    # a specific config file for the predicates
-    # TODO: maybe change this part to make a more customized way of getting the predicates configurations
-    ontology_f = open("ontology_dict.csv", 'r')
-    ontology_reader = csv.DictReader(ontology_f)
-    ontology_dict = {}
-    
-    # converting the reader dict to the predicate dictionary
-    for row in ontology_reader:
-        ontology_dict[row[list(row.keys())[0]]] = row[list(row.keys())[1]]
-                
-    return ontology_dict
+    with open("relations_mapping.json", 'r') as mapper:
+        # a specific config file for the predicates
+        # TODO: maybe change this part to make a more customized way of getting the predicates configurations
+        data = json.load(mapper)
+    return data
     
 # a main operation function
 def convert_csv_to_turtle():
-    # TODO: maybe change this one to accomadate for multiple files
     if len(sys.argv) != 2:
-        raise ValueError("Invalid number of input filename") 
-    
-    g = Graph()
-
-    csv_file = open(sys.argv[1], 'r', encoding='utf-8')
-    
-    csv_reader = csv.DictReader(csv_file)
-    ontology_dict = get_ontology_dict()
+            raise ValueError("Invalid number of input filename") 
+        
+    with open(sys.argv[1], 'r', encoding='utf-8') as csv_file:
+        # TODO: maybe change this one to accomadate for multiple files
 
-    # Convert each row to Turtle format and add it to the output
-    for row in csv_reader:
-        # the first column as the subject
-        absolute_uri = URIRef(row[list(row.keys())[0]])
-        g.add((absolute_uri, RDF.type, URIRef("http://www.wikidata.org/prop/direct/chant")))
+        g = Graph()
         
-        # extracting other informations
-        for k in row:
-            # if this key is the first column, skip
-            if k.__eq__(list(row.keys())[0]):
-                continue
-            
-            # finding the predicate from csv in the config dictionary, if not exit, skip
-            if k in list(ontology_dict.keys()):
-                predicate = URIRef(ontology_dict[k])
-            else:
-                continue
-            
-            # the object might be an URI or a literal
-            if validators.url(row[k]):
-                obj = URIRef(row[k])
-            else:
-                obj = Literal(row[k])
-            
-            # add this triple (absolute uri, predicate, object) to the graph
-            g.add((absolute_uri, predicate, obj))    
+        csv_reader = csv.reader(csv_file)
+        ontology_dict = get_ontology_dict()
         
-    csv_file.close()
-    
+        header = next(csv_reader)
+
+        # Convert each row to Turtle format and add it to the output
+        for row in csv_reader:
+            # the first column as the subject
+            key_attribute = URIRef(row[0])
+            if "type" in list(ontology_dict.keys()):
```

Comment:
If `type` is not in `ontology_dict` what happens to the first column of each row?  Is it just not added to the turtle file?

Reply:
Yes. The type attribute should be added to the mapper file. I am still considering what should be a consequence if the type is not in the mapper since it depends on the definitions and protocols of the TTL file.

Reply:
Also, you're checking this on every iteration of the loop, when you should be able to check and raise the `ValueError` in one place. If it passes, then you know that your ontology dict is complete.